### PR TITLE
 Add DB termination component on application shutdown

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
 Adds DB termination component which stops persistence service and optionally network cache coordination service  on application shutdown.

### CODE REVIEW

**Comment by mshaposhnik** [comment]
ci-build

File: core/che-core-db/src/main/java/org/eclipse/che/core/db/DBTermination.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+/**
+ * Stops {@link PersistService} when a system is ready to shutdown.
+ *
+ * @author Anton Korneta
+ */
+@Singleton
+public class DBTermination {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DBTermination.class);
```
Comment (sleshchenko) [inline core/che-core-db/src/main/java/org/eclipse/che/core/db/DBTermination.java:32]: I would add empty line to group static and non-static variables

File: core/che-core-db/src/main/java/org/eclipse/che/core/db/DBTermination.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  private PersistService persistService;
```
Comment (sleshchenko) [inline core/che-core-db/src/main/java/org/eclipse/che/core/db/DBTermination.java]: final?

File: core/che-core-db/src/main/java/org/eclipse/che/core/db/DBTermination.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    }
+  }
+
+  /**
+   * This method is hack that changes value of {@link JGroupsRemoteConnection#isLocal} to false.
+   * This is needed to close the JGroups EclipseLinkCommandChannel and as result gracefully stop of
+   * the system.<br>
+   * For more details see {@link JGroupsRemoteConnection#closeInternal()}
+   *
+   * <p>TODO eclipse-link extension issue https://bugs.eclipse.org/bugs/show_bug.cgi?id=534148
```
Comment (sleshchenko) [inline core/che-core-db/src/main/java/org/eclipse/che/core/db/DBTermination.java]: Looks like this TODO is redundant

File: core/che-core-db/src/main/java/org/eclipse/che/core/db/DBTermination.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  private PersistService persistService;
+  private EntityManagerFactory emFactory;
+
+  @Inject
+  public DBTermination(PersistService persistService, EntityManagerFactory emFactory) {
+    this.persistService = persistService;
+    this.emFactory = emFactory;
+  }
+
+  /** Stops {@link PersistService}. Any DB operations is impossible after that. */
```
Comment (sleshchenko) [inline core/che-core-db/src/main/java/org/eclipse/che/core/db/DBTermination.java]: I would say `Any DB operations are not permitted after that`

**Comment by mshaposhnik** [comment]
ci-build

**Review by sleshchenko — APPROVED** [review]

**Review by skabashnyuk — APPROVED** [review]

**Comment by sleshchenko** [comment]
ci-test

**Review by garagatyi — APPROVED** [review]

**Pull request merged**